unlock-app/src/services/walletService.js

```diff
@@ -91,12 +102,22 @@ export default class WalletService extends EventEmitter {
     }
   }
 
+  /**
+   * Poll to see if account has changed
+   */
+  async pollForAccountChange(timeout, isServer) {
+    if (isServer) return
+    this.account = await this.getAccount(true, this.account)
+    timeout(this.pollForAccountChange, POLLING_INTERVAL)
+  }
+
   /**
    * Function which yields the address of the account on the provider or creates a key pair.
    */
-  getAccount() {
+  getAccount(polling = false, priorAccount) {
```

Comment:
Hum, I think we should keep that function "clean" and put the logic of checking wether the account changed inside the `pollForAccountChange`? `getAccount` should just get the account while `pollForAccountChange` is actually the function which should know/care about whether the account changed.